### PR TITLE
Fix prefetcher concurrent map read and write issue

### PR DESCRIPTION
**Pull request by andyzhang2023**
### Description
It's a bug. The triePrefetcher structure should be thread safe, and it occasionally crashes the service with fatal error of "concurrent map read and write" on the map `fetchers`. We just add a lock to protected the map here.

### Changes
 the map `fetchers` will be protected by an RWMutex, and there are five function calls in which the map `fetchers` will be accessed:
* prefetch(): read and write, need to be Lock
* trie(): read only, need to be RLock
* copy(): read only, need to be RLock
* close(): read and wirte, need to be Lock
* used(): read only, need to be RLock

### CODE REVIEW

File: core/state/trie_prefetcher.go [hunk 1]
```diff
@@ -190,7 +225,11 @@ func (p *triePrefetcher) trie(owner common.Hash, root common.Hash) Trie {
 // used marks a batch of state items used to allow creating statistics as to
 // how useful or wasteful the prefetcher is.
 func (p *triePrefetcher) used(owner common.Hash, root common.Hash, used [][]byte) {
-	if fetcher := p.fetchers[p.trieID(owner, root)]; fetcher != nil {
+	p.fetchersLock.Lock()
```
Comment (welkin22) [inline core/state/trie_prefetcher.go:228]: The purpose of fetchersLock is to protect the fetchers map from concurrent read and write operations. In this case, fetchers is only being read and not written, so perhaps an RLock is sufficient.
As for the change `fetcher.used = used`, it should not be within the scope of the lock's protection.

Comment (andyzhang2023) [reply core/state/trie_prefetcher.go]: It sounds good! Applied.

**Review by welkin22 — APPROVED** [review]
LGTM

**Review by owen-reorg — APPROVED** [review]
LGTM

**Comment by owen-reorg** [comment]
close this issue for now since we removed the feature in the v0.3.0 release

**Pull request closed without merging**